src/Illuminate/Auth/TokenGuard.php

```diff
@@ -112,12 +112,7 @@ public function validate(array $credentials = [])
         }
 
         $credentials = [$this->storageKey => $credentials[$this->inputKey]];
-
-        if ($this->provider->retrieveByCredentials($credentials)) {
-            return true;
-        }
-
-        return false;
+        return (bool) $this->provider->retrieveByCredentials($credentials);
```

Comment:
It works but IMHO it solution is not a better then current.

If you really need to change something here, you can use
```return null !== $this->provider->retrieveByCredentials($credentials);```  for example, it covers both possible return values. Also it anyway works faster then typecasting.

Reply:
Good idea :+1:

Reply:
Implemented in https://github.com/rectorphp/rector/pull/780